### PR TITLE
treewide: drop Qt5 < 5.15

### LINKED ISSUE #201765: Drop Qt 5.12 and 5.14
We have 32 packages that depend on pre-5.15 Qt. Those are (as determined by `rg '\s+(.*) = libsForQt51[24]\.callPackage' --no-line-number -o -r '$1'`):

- [x] antimony (upstream, see https://github.com/mkeeter/antimony/pull/235)
- [x] apmplanner2 https://github.com/NixOS/nixpkgs/pull/203064
- [x] bitcoin-classic https://github.com/NixOS/nixpkgs/pull/202664
- [x] bitcoin-gold https://github.com/NixOS/nixpkgs/pull/202664
- [x] bitcoin-unlimited https://github.com/NixOS/nixpkgs/pull/202664
- [ ] boomerang
- [x] brewtarget https://github.com/NixOS/nixpkgs/pull/203064
- [x] cb2bib https://github.com/NixOS/nixpkgs/pull/203064
- [x] digibyte https://github.com/NixOS/nixpkgs/pull/202664
- [x] digitalbitbox https://github.com/NixOS/nixpkgs/pull/203064
- [ ] fontmatrix
- [ ] foxitreader
- [x] graphia https://github.com/NixOS/nixpkgs/pull/203057
- [x] ipe https://github.com/NixOS/nixpkgs/pull/203064
- [ ] kexi
- [x] kvirc #201771
- [x] mythtv https://github.com/NixOS/nixpkgs/pull/203145
- [x] notepadqq #202075
- [ ] olive-editor
- [x] polyphone #201974 
- [ ] pulseview (upstream, see https://github.com/NixOS/nixpkgs/issues/201765#issuecomment-1320084833)
- [ ] quiterss
- [x] rosegarden https://github.com/NixOS/nixpkgs/pull/203064
- [x] sayonara #202142
- [x] smtube: https://github.com/NixOS/nixpkgs/pull/201770
- [x] subsurface #202145
- [ ] swift-im
- [ ] tony
- [x] vertcoin https://github.com/NixOS/nixpkgs/pull/202664
- [x] wkhtmltopdf https://github.com/NixOS/nixpkgs/pull/203064
- [x] xygrib https://github.com/NixOS/nixpkgs/pull/203058

The old Qt versions are a big maintenance burden, as evidenced by them breaking yet again with https://github.com/NixOS/nixpkgs/pull/200872, and we should really get rid of them.

Ping @NixOS/qt-kde

Also general ping to all the maintainers of affected packages:

- antimony: @rnhmjoj
- apmplanner2: @wucke13
- bitcoin-classic: @jefdaj
- bitcoin-gold: @mmahut
- bitcoin-unlimited: @DmitryTsygankov
- boomerang: @dtzWill
- brewtarget: @mmahut
- cb2bib: @edwtjo
- digibyte: @mmahut
- digitalbitbox: @vidbina
- foxitreader: @p-h, @rhoriguchi
- graphia: @bgamari
- ipe: @ttuegel
- kexi: @zraexy
- kvirc: @suhr
- mythtv: @titanous
- notepadqq: @rszibele
- olive-editor: @balsoft
- polyphone: @Maxdamantus
- pulseview: @bjornfor
- quiterss: @primeos
- rosegarden: @lebastr
- sayonara: @deepfire
- smtube: @vbgl
- subsurface: @mguentner, @adisbladis
- swift-im: @orivej
- tony: @orivej
- vertcoin: @mmahut
- wkhtmltopdf: @jb55
- xygrib: @johannesloetzsch, @SuperSandro2000

### PR DESCRIPTION
**Pull request by K900**
###### Description of changes

Closes #201765.

I fully intend to merge this in the coming days unless there are any major objections.

List of affected packages:
| Package | Dead upstream | Notes | Maintainer pings |
|-|-|-|-|
| `airwave` | ? | unconfirmed, can't find a way to obtain required source | @michalrus |
| `boomerang` | [very](https://github.com/BoomerangDecompiler/boomerang) || @dtzWill |
| `foxitreader` | ? | can't fetch the source, package hasn't been updated for over a year | @p-h @rhoriguchi  |
| `kexi` | [basically](https://invent.kde.org/office/kexi) || @zraexy |
| `olive-editor` | ? | upstream is [active](https://github.com/olive-editor/olive), but last numbered release in 2019 - update to unstable? | @balsoft |
| `pulseview` | ? | deadness of upstream needs confirmation | @bjornfor |
| `quiterss` | [yes](https://github.com/QuiteRSS/quiterss/commits/master) | builds on 5.15 but segfaults on startup (might be Wayland specific?) | @primeos |
| `swift-im` | [very](https://github.com/swift/swift/) || @orivej |
| `tony` | [very](https://code.soundsoftware.ac.uk/projects/tony/repository) || @orivej |

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by K900** [comment]
Result of `nixpkgs-review pr 202661` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages failed to build:</summary>
  <ul>
    <li>bitcoin-classic</li>
    <li>bitcoin-gold</li>
    <li>boomerang</li>
    <li>digibyte</li>
    <li>graphia</li>
    <li>olive-editor</li>
    <li>pulseview</li>
    <li>tony</li>
    <li>vertcoin</li>
    <li>xygrib</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>bitcoin-unlimited</li>
    <li>common-updater-scripts</li>
    <li>digitalbitbox</li>
    <li>ipe</li>
    <li>rosegarden</li>
  </ul>
</details>

**Comment by K900** [comment]
Result of `nixpkgs-review pr 202661` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages failed to build:</summary>
  <ul>
    <li>bitcoin-classic</li>
    <li>bitcoin-gold</li>
    <li>boomerang</li>
    <li>digibyte</li>
    <li>foxitreader</li>
    <li>graphia</li>
    <li>kexi</li>
    <li>mythtv</li>
    <li>odoo</li>
    <li>olive-editor</li>
    <li>pulseview</li>
    <li>swift-im</li>
    <li>tony</li>
    <li>vertcoin</li>
    <li>xygrib</li>
  </ul>
</details>
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>apmplanner2</li>
    <li>bitcoin-unlimited</li>
    <li>brewtarget</li>
    <li>cb2bib</li>
    <li>common-updater-scripts</li>
    <li>digitalbitbox</li>
    <li>fontmatrix</li>
    <li>ipe</li>
    <li>quiterss</li>
    <li>rosegarden</li>
    <li>wkhtmltopdf</li>
  </ul>
</details>

**Comment by K900** [comment]
Updated including broken/insecure/unfree packages.

**Comment by K900** [comment]
Also, I will flag everything that FTBFS right before merging the PR. Not doing it now to simplify potential fix merges.

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -792,6 +792,9 @@ mapAliases ({
   librsync_0_9 = throw "librsync_0_9 has been removed"; # Added 2021-07-24
   librewolf-wayland = librewolf; # Added 2022-11-15
   libseat = seatd; # Added 2021-06-24
+  libsForQt512 = throw "Qt 5 versions prior to 5.15 are no longer supported upstream";  # Added 2022-11-24
+  libsForQt514 = throw "Qt 5 versions prior to 5.15 are no longer supported upstream";  # Added 2022-11-24
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:796]:
```suggestion
  libsForQt512 = throw "Qt 5 versions prior to 5.15 are no longer supported upstream and have been removed.";  # Added 2022-11-24
  libsForQt514 = throw "Qt 5 versions prior to 5.15 are no longer supported upstream and have been removed.";  # Added 2022-11-24
```

just to make it clear why the error appears

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -1266,6 +1269,9 @@ mapAliases ({
   qr-filetransfer = throw ''"qr-filetransfer" has been renamed to "qrcp"''; # Added 2020-12-02
   qshowdiff = throw "'qshowdiff' (Qt4) is unmaintained and not been updated since its addition in 2010"; # Added 2022-06-14
   qt-3 = throw "qt-3 has been removed from nixpkgs, as it's unmaintained and insecure"; # Added 2021-02-15
+  qt512 = throw "Qt 5 versions prior to 5.15 are no longer supported upstream";  # Added 2022-11-24
+  qt514 = throw "Qt 5 versions prior to 5.15 are no longer supported upstream";  # Added 2022-11-24
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:1273]: same here

**Review by thiagokokada — APPROVED** [review]

**Review by balsoft — APPROVED** [review]

File: pkgs/top-level/aliases.nix (hunk 1 above)
Comment (K900) [reply pkgs/top-level/aliases.nix]: Thanks, applied.

**Comment by rhoriguchi** [comment]
Foxitreader is dead, there's not even a download link listed on their web page. Package download [link](https://cdn01.foxitsoftware.com/pub/foxit/reader/desktop/linux/2.x/2.4/en_us/FoxitReader.enu.setup.2.4.4.0911.x64.run.tar.gz) and package still work.

**Review by rhoriguchi — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by K900** [comment]
Let's go.

**Comment by zowoq** [comment]
It was caused by a mistake in another PR (https://github.com/NixOS/nixpkgs/pull/202465#issuecomment-1336114733) but should have waited for CI to finish after the force-push as it would have shown that merging this would break eval.

**Comment by K900** [comment]
That's what I get for trusting git too much, I guess. Thanks for the fix!

**Comment by meck** [comment]
Pulseview repo master branch at git://sigrok.org/pulseview.git has some commits for enabling building with qt6, i havent had time to test @bjornfor

**Pull request merged**